src/main/configurations/xxllnc/Configuration_Handler_AddDocumentsToCaseCommand.xml

```diff
@@ -32,25 +32,29 @@
                 parallel="true"
                 collectResults="true"
                 >
-                <IbisLocalSender
-                    javaListener="Handler_AddDocumentToCaseCommand"
-                    >
-                    <Param name="url" xpathExpression="/url" />
-                    <Param name="messageId" xpathExpression="concat($CaseReferenceNumber, '-', tokenize(/url, '/')[last()])">
+                    <IbisLocalSender
+                        javaListener="Handler_AddDocumentToCaseCommand"
+                        >
+                        <Param name="url" xpathExpression="/url" />
+                        <Param name="messageId" xpathExpression="concat($CaseReferenceNumber, '-', tokenize(/url, '/')[last()])">
+                            <Param name="CaseReferenceNumber" sessionKey="originalMessage" xpathExpression="/*/CaseReferenceNumber" />
```

Comment:
I believe correlationId doesn't have to be passed by param. If I remember correctly, sessionkey "cid" should be available in the adapter already. Any listener automatically adds it.